### PR TITLE
Remove gaugeToPoolAddress call in Gauges state

### PR DESCRIPTION
**Pull request by nguyenaiden**
***Description***: In preparation for retrieving gauges while on side chain, we're aiming to reduce the reliance on `gaugeHelperContract` as it doesn't exist on side chain. We no longer need `gaugeHelperContract.gaugeToPoolAddress()` to match a gauge to a pool. We now use the `BasicPools` context and retrieve the information from there.  With the current state on mainnet, this will reduce 1 large multicall step.

### CODE REVIEW

File: src/utils/gauges.ts [hunk 1]
```diff
@@ -232,8 +238,9 @@ export async function getGaugeData(
           gaugeName: gaugeNames[index],
           lpTokenAddress,
           isKilled: gaugeKillStatuses[index] ?? false,
-          poolAddress: isValidPoolAddress && poolAddress ? poolAddress : null,
-          poolName: gaugePool?.poolName || null,
+          poolAddress:
+            isValidPoolAddress && pool?.poolAddress ? pool?.poolAddress : null,
```
Comment (ug02fast) [inline src/utils/gauges.ts:242]: does it make sense to early return here? you should also be able to do `poolAddress: isValidPoolAddress && pool?.poolAddress` to shorten logic

**Review by ug02fast — APPROVED** [review]
lgtm just one observation

File: src/utils/gauges.ts [hunk 2]
```diff
@@ -198,19 +191,32 @@ export async function getGaugeData(
...
       (previousGaugeData, gaugeAddress, index) => {
         const lpTokenAddress = gaugeLpTokenAddresses[index]?.toLowerCase()
-        const poolAddress = gaugePoolAddresses[index]
+        const pool = lpTokenToPool[lpTokenAddress]
         const isValidPoolAddress = Boolean(
-          poolAddress && !isAddressZero(poolAddress),
-        )
-        const gaugePool = Object.values(basicPools || {}).find(
-          (pool) => pool.poolAddress === poolAddress,
+          pool && pool.poolAddress && !isAddressZero(pool.poolAddress),
```
Comment (ug02fast) [inline src/utils/gauges.ts:219]: you should be able to shorten this to `pool?.poolAddress && ...`

Comment (hammeiam) [reply src/utils/gauges.ts:219]: agreed, I don't think poolAddress would ever be 0x0 here

Comment (hammeiam) [reply src/utils/gauges.ts:219]: Furthermore since you just created that map and you know that each key has a valid value, you can just say `const pool = lpTokenToPool[lpTokenAddress] as MyPoolType` so you don't have to do the null checks later on

File: src/utils/gauges.ts (hunk 1 above)
Comment (hammeiam) [reply src/utils/gauges.ts:242]: agreed

File: src/utils/gauges.ts (hunk 2 above)
Comment (nguyenaiden) [reply src/utils/gauges.ts:219]: @ug02fast, I'm running into the issue of having to check for both `undefined` as well as `null` types. This is mainly coming from our need to use `?` optional chaining, but the `.tryAll` method is returning `Promise<null>` if the call fails.

What do you think about checking for both, or we pick 1 type of falsy value and go with that?

**Review by ug02fast — APPROVED** [review]
